### PR TITLE
Update AWS SNS integration to not use grafana\aws-sdk-go

### PR DESCRIPTION
**Pull request by yuri-tceretian**
The integration was introduced in https://github.com/grafana/alerting/pull/173.
It imported a new module to alerting - `github.com/grafana/grafana-aws-sdk`. That module has quite a long list of dependencies, which makes all modules that use alerting pull all its dependencies. 

As long as the alerting module is vendored in Mimir project, the introduction of the Grafana's AWS SDK is sub-optmial. The alerting team decided to avoid unnecessary imports and update the integration to not use the module. 

This PR updates SNS integration with the logic copied from the upstream implementation. There is only one notable difference - because Grafana integrations use abstract webhook client, the SNS client is set up to use default HTTP client. The upstream version uses client configured for the integration https://github.com/prometheus/alertmanager/blob/33e680a34dc2e54508b9fcdb2b8c0ee81fd61d87/notify/sns/sns.go#L139C15-L139C80

cc @robbierolin

### CODE REVIEW

**Comment by robbierolin** [comment]
The grafana-aws-sdk has logic for assuming roles in opt-in regions that I don't think would be supported with this implementation
https://github.com/grafana/grafana-aws-sdk/blob/main/pkg/awsds/sessions.go#L169
And I think some other settings that make things consistent across aws data sources - https://github.com/grafana/grafana-aws-sdk/blob/main/pkg/awsds/authSettings.go#L43

**Comment by yuri-tceretian** [comment]
@robbierolin thanks for the feedback. Our main course is to unify Grafana and Prometheus integrations, and eventually get rid of the former wherever it's possible. I think going with Prometheus' implementation in this case justifies some regression in the supported feature set.

File: receivers/sns/config_test.go [hunk 1]
```diff
@@ -76,9 +91,13 @@ func TestNewConfig(t *testing.T) {
 			settings: `{
 				"topic_arn": "arn:aws:sns:region:0123456789:SNSTopicName"
 			}`,
-			expectedAuthType: awsds.AuthTypeDefault.String(),
-			expectedSubject:  templates.DefaultMessageTitleEmbed,
-			expectedMessage:  templates.DefaultMessageEmbed,
+			expected: Config{
+				APIUrl:   "https://sns..amazonaws.com",
```
Comment (stevesg) [inline receivers/sns/config_test.go]: Typo or intentional?

File: receivers/sns/config_test.go [hunk 2]
```diff
@@ -87,16 +106,33 @@ func TestNewConfig(t *testing.T) {
 				"subject": "subject",
 				"message": "message"
 			}`,
-			expectedSubject:  "subject",
-			expectedMessage:  "message",
-			expectedAuthType: awsds.AuthTypeDefault.String(),
+			expected: Config{
+				APIUrl:   "https://sns..amazonaws.com",
```
Comment (stevesg) [inline receivers/sns/config_test.go]: Typo or intentional?

File: receivers/sns/sns.go [hunk 3]
```diff
@@ -21,61 +23,58 @@ import (
...
+
+	// check template error after we use them
+	if tmplErr != nil {
+		s.log.Warn("failed to template message", "error", tmplErr.Error())
 	}
 
-	snsClient := sns.New(session, aws.NewConfig().WithEndpoint(*aws.String(s.settings.APIUrl)))
 	publishOutput, err := snsClient.Publish(publishInput)
 	if err != nil {
 		s.log.Error("Failed to publish to Amazon SNS. ", "error", err)
```
Comment (stevesg) [inline receivers/sns/sns.go:66]: We want to return the error here. Should we check the error for whether we should retry or not, like upstream does?

```suggestion
		s.log.Error("Failed to publish to Amazon SNS. ", "error", err)
		return true, err
```

File: receivers/sns/sns.go [hunk 4]
```diff
@@ -21,61 +23,58 @@ import (
...
 }
 
-func (s *Notifier) createPublishInput(ctx context.Context, as ...*types.Alert) (*sns.PublishInput, error) {
-	var err error
-	tmpl, _ := templates.TmplText(ctx, s.tmpl, as, s.log, &err)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create template: %w", err)
-	}
-
+func (s *Notifier) createPublishInput(ctx context.Context, tmpl func(string) string) (*sns.PublishInput, error) {
```
Comment (stevesg) [inline receivers/sns/sns.go]: [mega-nitpick] If this function was under `createSNSClient`, the code would be easier to compare to upstream. But this is a total nit, feel free to ignore.

**Review by stevesg — COMMENTED** [review]
LGTM except one place I think we need a `return`. Reviewed mainly by comparing to upstream code; not tested.

File: receivers/sns/config_test.go (hunk 2 above)
Comment (yuri-tceretian) [reply receivers/sns/config_test.go]: That's the logic of the integration.

Comment (yuri-tceretian) [reply receivers/sns/config_test.go]: I will change it to match the upstream

**Review by stevesg — APPROVED** [review]

**Pull request merged**